### ISSUE TITLE
model replication

### ISSUE THREAD

**Issue by coisino** [Jianf-Wang/NP-SemiSeg#2]
Hello, I'm reproducing a model using the VOC2012 dataset in AugSeg, and the test results I obtained have very confusing predicted labels. I can't even discern the rough shape of any object. What could be the reason for this?

**Comment by Jianf-Wang** [comment]
Hi, thanks for your message. Did you successfully train the model ? what are mIOU scores on the test set (it would be great if you could provie the training log) ?  If your model achieves good performance on the test set, you may check your visualization code, since  I did not plot maps for the AugSeg framework.

**Comment by coisino** [comment]
Hello, I trained for 80 epochs and then incorporated the visualization part from U2PL into AugSeg. These two files are the training logs and the files generated during testing.
[val_best_20230917_102550.txt](https://github.com/Jianf-Wang/NP-SemiSeg/files/12641918/val_best_20230917_102550.txt)
[zsing_run_voc.log](https://github.com/Jianf-Wang/NP-SemiSeg/files/12641919/zsing_run_voc.log)

The visualization results I obtained look like this.
![2007_000033](https://github.com/Jianf-Wang/NP-SemiSeg/assets/48575685/f6d21761-38b9-4fdd-958b-cd3e17459647)

**Comment by Jianf-Wang** [comment]
Your model is trained well on VOC2012 (based on your training log), as it achieves good mIOU values on the test set. Maybe you need to check your visualization code carefully (e.g., whether the trained model is loaded properly). Plus, you may want to check whether the trained model is saved during training.

**Comment by coisino** [comment]
I am training with the U2PL model, but there is an issue with the output information: Uns 0.000 (0.000) Con 0.000 (0.000). What could be the reason for this?

**Comment by coisino** [comment]
I trained using the U2PL model, and the visualization results from the tests are normal. The results from the model trained for 2 epochs are acceptable. However, the visualization results are incorrect after training for 80 epochs on AugSeg.

**Comment by Jianf-Wang** [comment]
Hi, I did not do visualization with AugSeg. I directly used results from the training log file when writing the paper. If you want to do visualization with AugSeg, you need to write the testing code and carefully fix the training code by yourself. For example, the training code for AugSeg doesn't save memory banks during training (see lines 227-233 in "train_semi.py").

**Comment by coisino** [comment]
May I have a look at your U2PL training log ? The value of the unsupervised loss ( Uns ) in the training log that I now get is always close to zero after the first epoch ( Data 0.00 ( 0.00 ) Time 0.80 ( 0.83 ) Sup 0.091 ( 0.154 ) Uns 0.006 ( 0.010 ) Con 5.076 ( 4.931 ) LR 0.00098 )

**Comment by Jianf-Wang** [comment]
As the training progresses, U2PL will calculate the unsupervised loss. This can be changed by modifying configuration files.
